pkg/ipamd/datastore/data_store_test.go

```diff
@@ -235,6 +365,143 @@ func TestDelENIIPv4Address(t *testing.T) {
 	assert.Equal(t, len(ds.eniPool["eni-1"].AvailableIPv4Cidrs), 1)
 }
 
+func TestDelENIIPv4AddressWithPDEnabled(t *testing.T) {
```

Comment:
nit: this also feels like it should be calling `TestDelENIIPv4Address` and passing a PD boolean

Reply:
Same as above, to avoid conditional checks.